front/components/assistant/conversation/AgentMessage.tsx

```diff
@@ -311,6 +341,77 @@ export function AgentMessage({
       }
     );
   }
+  function Citations({
+    activeReferences,
+  }: {
+    activeReferences: { index: number; document: RetrievalDocumentType }[];
+  }) {
+    const citationContainer = useRef<HTMLDivElement>(null);
+
+    useEffect(() => {
+      if (citationContainer.current && lastHoveredReference !== null) {
+        // get the offset of the first child in its container
+        const offset = (
+          citationContainer.current.firstElementChild
+            ?.firstElementChild as HTMLElement
+        ).offsetLeft;
+        const scrolling =
+          (citationContainer.current.firstElementChild?.firstElementChild
+            ?.scrollWidth || 0) *
+          (lastHoveredReference - 2);
+        citationContainer.current.scrollLeft = scrolling - offset;
+      }
+    });
+
+    activeReferences.sort((a, b) => a.index - b.index);
+    return (
+      <div
+        className="-mx-[100%] mt-9 overflow-x-auto px-[100%] scrollbar-hide"
+        id={`message-${message.sId}-citations`}
+        ref={citationContainer}
+      >
+        <div className="left-100 relative flex gap-2">
+          {activeReferences.map(({ document, index }) => {
+            const provider = providerFromDocument(document);
+            return (
+              <div
+                className="flex w-48 flex-none flex-col gap-2 rounded-xl border border-structure-100 p-3 sm:w-64"
+                key={index}
+              >
+                <div className="flex items-center gap-1.5">
+                  <div className="flex h-5 w-5 items-center justify-center rounded-full border border-violet-200 bg-violet-100 text-xs font-semibold text-element-800">
```

Comment:
The numbers are not perfectly centered :grimacing: 
![Screenshot from 2023-10-10 12-21-51](https://github.com/dust-tt/dust/assets/15067/58ab5cb8-6303-498c-8ac3-90326ec77f8a)

Reply:
Do you have context on when it's not centered? perfectly centered on my end
![image](https://github.com/dust-tt/dust/assets/5437393/a413ee2d-d870-4885-bfd5-0b86509f8ec2)

Reply:
huh....